tests/unit/test_stages/base.py

```diff
@@ -96,6 +100,24 @@ def test_stage_state(self, review_stage_state, visited, expected_stage_state):
 
             self.assertEqual(self.block.get_stage_state(), expected_stage_state)
 
+    @ddt.data(
+        (datetime(2015, 11, 11), datetime(2015, 11, 11), True, False),
+        (datetime(2015, 11, 9), datetime(2015, 11, 12), True, False),
+        (datetime(2015, 11, 10), datetime(2015, 11, 10), True, True),
+        (datetime(2015, 11, 9), datetime(2015, 11, 10), True, True),
+        (datetime(2015, 11, 9), datetime(2015, 11, 12), True, False),
+        (None, datetime(2015, 11, 12), True, False),
+        (None, None, True, False),
+        (None, datetime(2015, 11, 10), True, True),
```

Comment:
@jbzdak nit: it would be good to give some explanations on what test means what - like in integration/tests_stages -> test_open_close_label